integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/TopicBasicsTestCase.java

```diff
@@ -0,0 +1,159 @@
+     * 5. Receive them by two subscribers.
+     * @throws JMSException
+     * @throws NamingException
+     * @throws IOException
+     */
+    @Test(groups = "wso2.mb.topic")
+    public void performTopicSendReceive() throws JMSException, NamingException, IOException,
+            InterruptedException {
+
+        TopicSender sender = null;
+        TopicReceiver subscriber = null;
+        TopicReceiver subscriber2 = null;
+        try {
+            subscriber = new TopicReceiver("testTopic", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, configurationReader,
+                    null);
+
+            subscriber2 = new TopicReceiver("testTopic", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, configurationReader,
+                    null);
+
+            //send a message
+            sender = new TopicSender("testTopic", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, configurationReader);
+            sender.sendMessages(1, "test message");
+
+            Assert.assertNotNull(subscriber.receiveMessage(ClientConstants.CLIENT_WAIT_DURATION),
+                    "Topic not received by the subscriber");
+            Assert.assertNotNull(subscriber2.receiveMessage(ClientConstants.CLIENT_WAIT_DURATION),
+                    "Topic not received by the subscriber2");
+
+            //send 500 messages
+            sender.sendMessages(500, "test message");
+            Assert.assertEquals(subscriber.getMessageCount(), 500, "Did not receive all the topics by "
+                    + "subscriber1");
+            Assert.assertEquals(subscriber2.getMessageCount(), 500, "Did not receive all the topics by "
+                    + "subscriber2");
+        } finally {
+            if (sender != null) {
+                sender.closeSender();
+                subscriber.closeReceiver();
+            }
+        }
+    }
+
+    /**
+     * Test for durable subscription.
+     * 1. Create two durable subscribers.
+     * 2. Send 10 messages and check for their reception.
+     * 3. Close one subscriber.
+     * 4. Send 10 messages.
+     * 5. Start the closed subscriber again.
+     * 6. Check the reception of messages.
+     * @throws JMSException
+     * @throws NamingException
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    @Test(groups = "wso2.mb.topic")
+    public void testDurableSubscription() throws JMSException, NamingException, IOException, InterruptedException {
+        TopicSender sender = null;
+        TopicReceiver subscriber = null;
+        TopicReceiver subscriber2 = null;
+        try {
+            subscriber = new TopicReceiver("testTopic2", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, configurationReader,
+                    "sub1");
+
+            subscriber2 = new TopicReceiver("testTopic2", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, configurationReader,
+                    "sub2");
+
+
+            //send 10 messages
+            sender = new TopicSender("testTopic2", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, configurationReader);
```

Comment:
Shall we make all the queue names unique ones that describe the test methods that are being run. That way, there will not be any confusions when a test fails. Otherwise if 1 messages remains in the queue dues to a failure in a single test method the rest of the tests which use the same queue will fail for no reason.

Reply:
Will change.